compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -601,6 +601,12 @@ impl<'a> Parser<'a> {
         self.last_unexpected_token_span = Some(self.token.span);
         let mut err = self.struct_span_err(self.token.span, &msg_exp);
 
+		if let TokenKind::Ident(symbol, _) = &self.prev_token.kind {
+			if symbol.to_string() == "public" {
```

Comment:
```suggestion
			if symbol.as_str() == "public" {
```
nit: avoid going through `ToString`